File: 0649-dota2-senate/0649-dota2-senate.py
class Solution(object):
    def predictPartyVictory(self, senate):
        """
        :type senate: str
        :rtype: str
        """ 
        queueR=[]
        queueD=[]
        l=len(senate)
        for i in range(l):
            if senate[i] == 'R':
                queueR+=[i]
            else:
                queueD+=[i]
        
        while queueR and queueD:
            indexR=queueR[0]
            indexD=queueD[0]
            queueR.pop(0)
            queueD.pop(0)            
            if indexR < indexD:
                queueR+=[indexR+l]
            else:
                queueD+=[indexD+l]
        
        if len(queueR)>len(queueD):
            return "Radiant"
        return "Dire"

        

File: 0649-dota2-senate/0649-dota2-senate.cpp
class Solution {
public:
    string predictPartyVictory(string senate) {
        int numR{0},numD{0};
        char active = 0;
        int activeCount=0;
        while(senate.length()>0)
        {
            bool changed=false;
            
            string nextRound="";
            for(char c: senate)
                if(!active)
                {
                    active=c;
                    nextRound+=c;
                    activeCount++;
                }
                else{
                    if(active==c)
                        nextRound+=c,activeCount++;
                    else
                    {
                        changed=true;
                        activeCount--;
                        if(activeCount==0)
                            active=0;

                    }
                }
            
            if(!changed)
                if(active=='R')
                    return "Radiant";
                else
                    return "Dire";
            senate=nextRound;            
        }
        return "";
    }
};